Index: oce/src/Draw/Draw_Window.cxx
===================================================================
--- oce.orig/src/Draw/Draw_Window.cxx
+++ oce/src/Draw/Draw_Window.cxx
@@ -31,6 +31,8 @@
 #include <TCollection_AsciiString.hxx>
 #include <Image_AlienPixMap.hxx>
 
+#include <stdio.h>
+
 extern Draw_Interpretor theCommands;
 extern Standard_Boolean Draw_VirtualWindows;
 static Tcl_Interp *interp;        /* Interpreter for this application. */
@@ -134,7 +136,6 @@ defaultPrompt:
 # include <strings.h>
 #endif
 
-#include <stdio.h>
 #include <tk.h>
 
 /*
