Index: oce/inc/V3d_View.hxx
===================================================================
--- oce.orig/inc/V3d_View.hxx
+++ oce/inc/V3d_View.hxx
@@ -857,7 +857,7 @@ public:
 //!  view on level of OpenGl_GraphicDriver; Please take it into account if <br>
 //!  you use it for your purposes; <br>
 //!  Warning: Works only under Windows. <br>
-  Standard_EXPORT     Standard_Boolean Print(const Aspect_Handle hPrnDC = 0,const Standard_Boolean showDialog = Standard_True,const Standard_Boolean showBackground = Standard_True,const Standard_CString filename = NULL,const Aspect_PrintAlgo printAlgorithm = Aspect_PA_STRETCH) const;
+  Standard_EXPORT     Standard_Boolean Print(const Aspect_Handle hPrnDC = OCE_NULL_Aspect_Handle,const Standard_Boolean showDialog = Standard_True,const Standard_Boolean showBackground = Standard_True,const Standard_CString filename = NULL,const Aspect_PrintAlgo printAlgorithm = Aspect_PA_STRETCH) const;
   //! Dumps the full contents of the view <br>
 //!           to a pixmap of pixel size <theWidth> * <theHeight> and <br>
 //!           buffer type <theBufferType>. If <theToKeepAspect> is true <br>
Index: oce/src/Aspect/Aspect_Handle.hxx
===================================================================
--- oce.orig/src/Aspect/Aspect_Handle.hxx
+++ oce/src/Aspect/Aspect_Handle.hxx
@@ -38,4 +38,6 @@ const Handle(Standard_Type)& STANDARD_TY
 /*============================================================================*/
 #endif
 
+#define OCE_NULL_Aspect_Handle ((Aspect_Handle)NULL)
+
 #endif /* _Aspect_Handle_HeaderFile */
Index: oce/src/V3d/V3d_View.cdl
===================================================================
--- oce.orig/src/V3d/V3d_View.cdl
+++ oce/src/V3d/V3d_View.cdl
@@ -1451,7 +1451,7 @@ is
         --          extension must be one of ".png",".bmp",".jpg",".gif".
         --          Returns FALSE when the dump has failed
 
-      Print (me; hPrnDC: Handle from Aspect = 0;
+      Print (me; hPrnDC: Handle from Aspect = OCE_NULL_Aspect_Handle;
              showDialog: Boolean = Standard_True;
              showBackground : Boolean = Standard_True;
              filename: CString = NULL;
