From 58f9f0a527899f66817b98ceea278a0c5ff74fad Mon Sep 17 00:00:00 2001
From: Jerome Robert <jeromerobert@users.sf.net>
Date: Tue, 18 Dec 2012 15:49:45 +0100
Subject: [PATCH] Ensure GeomLib::NormEstim won't last for ever

---
 src/GeomLib/GeomLib.cxx |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

Index: oce/src/GeomLib/GeomLib.cxx
===================================================================
--- oce.orig/src/GeomLib/GeomLib.cxx
+++ oce/src/GeomLib/GeomLib.cxx
@@ -2399,7 +2399,14 @@ Standard_Integer GeomLib::NormEstim(cons
 
   gp_Vec NAux = DU^DV;
   Standard_Real h1 = h;
-  while(NAux.SquareMagnitude() < aTol2) {
+
+  //Ensure that the following loop won't last for
+  //ever. If after 50 iterations the normal is still
+  //bad we just give up
+  int nbiter = 0;
+  int maxIter = 50;
+
+  while(NAux.SquareMagnitude() < aTol2 && nbiter <= maxIter) {
     h1 += h;
     if(AlongV) {
       Standard_Real v = UV.Y() + sign*h1;
@@ -2417,8 +2424,11 @@ Standard_Integer GeomLib::NormEstim(cons
 
     }
     NAux = DU^DV;
+    nbiter++;
   }
 
+  if(nbiter == maxIter)
+    return 3;
 
   Iso->D2(t, DummyPnt, Tang, DD);
 
