Index: oce/src/Aspect/Aspect_MarkerStyle.cxx
===================================================================
--- oce.orig/src/Aspect/Aspect_MarkerStyle.cxx
+++ oce/src/Aspect/Aspect_MarkerStyle.cxx
@@ -532,6 +532,8 @@ void Aspect_MarkerStyle::SetPredefinedSt
 		  }
 		}
 		break ;
+		default :
+		break;
 	}
 
 }
Index: oce/src/BOPAlgo/BOPAlgo_Builder_3.cxx
===================================================================
--- oce.orig/src/BOPAlgo/BOPAlgo_Builder_3.cxx
+++ oce/src/BOPAlgo/BOPAlgo_Builder_3.cxx
@@ -306,6 +306,8 @@ void BOPAlgo_Builder::FillIn3DParts
     aSelector.Clear();
     aSelector.SetBox(aBoxS);
     //
+    aBBTree.Select(aSelector);
+    //
     const BOPCol_ListOfInteger& aLIFP=aSelector.Indices();
     //
     // 2.5. Collect faces that are IN aSolid [ aLFIN ]
Index: oce/src/BRepAlgo/BRepAlgo_Tool.cxx
===================================================================
--- oce.orig/src/BRepAlgo/BRepAlgo_Tool.cxx
+++ oce/src/BRepAlgo/BRepAlgo_Tool.cxx
@@ -68,7 +68,7 @@ TopoDS_Shape BRepAlgo_Tool::Deboucle3D(c
       // iterate on sub-shapes and add non-empty.
       TopoDS_Iterator it(S);
       TopoDS_Shape SubShape;
-      Standard_Boolean NbSub = 0;
+      Standard_Boolean NbSub = Standard_False;
       BRep_Builder B;
       if (S.ShapeType() == TopAbs_COMPOUND) {
 	B.MakeCompound(TopoDS::Compound(SS));
@@ -81,10 +81,10 @@ TopoDS_Shape BRepAlgo_Tool::Deboucle3D(c
 	SubShape = Deboucle3D(CurS,Boundary);
 	if ( !SubShape.IsNull()) {
 	  B.Add(SS, SubShape);
-	  NbSub++;
+	  NbSub = Standard_True;
 	}
       }
-      if (NbSub == 0)
+      if (!NbSub)
         {
 #ifdef DEB
         cout << "No subhape in shape!" << endl;
Index: oce/src/Draw/Draw.cxx
===================================================================
--- oce.orig/src/Draw/Draw.cxx
+++ oce/src/Draw/Draw.cxx
@@ -407,7 +407,7 @@ void Draw_Appli(Standard_Integer argc, c
         cin.get(cmd[++i]);
       } while ((cmd[i] != '\n') && (!cin.fail()));
       cmd[i] = '\0';
-    } while (Draw_Interprete(cmd) != (unsigned int ) -2);
+    } while (Draw_Interprete(cmd));
   }
 #ifdef WNT
   // Destruction de l'application
Index: oce/src/DrawResources/TestCommands.tcl
===================================================================
--- oce.orig/src/DrawResources/TestCommands.tcl
+++ oce/src/DrawResources/TestCommands.tcl
@@ -367,7 +367,10 @@ proc testgrid {args} {
         # commant to run DRAW with a command file;
         # note that empty string is passed as standard input to avoid possible 
         # hang-ups due to waiting for stdin of the launching process
-	set command "exec <<{} DRAWEXE -f $logdir/$group/$grid/${casename}.tcl"
+	if { [catch {set nameofexecutable [info nameofexecutable]}] } {
+		set nameofexecutable "DRAWEXE"
+	}
+	set command "exec <<{} $nameofexecutable -f $logdir/$group/$grid/${casename}.tcl"
 
 	# alternative method to run without temporary file; disabled as it needs too many backslashes
 #	else {
@@ -740,6 +743,13 @@ proc testfile {filelist} {
 proc locate_data_file {filename} {
     global env groupname gridname casename
 
+    if [info exists env(OCE_SKIP_TESTS_DATA)] {
+        # raise error
+        error [join [list "File $filename could not be found" \
+                          "(should be in paths indicated by CSF_TestDataPath environment variable, " \
+                          "or in subfolder data in the script directory)"] "\n"]
+    }
+
     # check if the file is located in the subdirectory data of the script dir
     set scriptfile [info script]
     if { $scriptfile != "" } {
Index: oce/src/FSD/FSD_CmpFile.cdl
===================================================================
--- oce.orig/src/FSD/FSD_CmpFile.cdl
+++ oce/src/FSD/FSD_CmpFile.cdl
@@ -319,7 +319,6 @@ is
 
     MagicNumber(myclass)
     	returns CString from Standard is private;
-    ---C++: return const
     
     RaiseError(me: in out ; theFailure: Type from Standard) is private;
         
Index: oce/src/Graphic3d/Graphic3d.cxx
===================================================================
--- oce.orig/src/Graphic3d/Graphic3d.cxx
+++ oce/src/Graphic3d/Graphic3d.cxx
@@ -25,6 +25,7 @@
 //function : InitGraphicDriver
 //purpose  :
 //=======================================================================
+#if !defined(OCE_BUILD_STATIC_LIB) && !defined(HAVE_NO_DLL)
 Handle(Graphic3d_GraphicDriver) Graphic3d::InitGraphicDriver (const Handle(Aspect_DisplayConnection)& theDisplayConnection)
 {
 #if !defined(_WIN32) && !defined(__WIN32__) && (!defined(__APPLE__) || defined(MACOSX_USE_GLX))
@@ -98,3 +99,15 @@ Handle(Graphic3d_GraphicDriver) Graphic3
 
   return aGraphicDriver;
 }
+#else
+
+extern "C"
+{
+     Handle(Graphic3d_GraphicDriver) MetaGraphicDriverFactory(const Standard_CString AShrName);
+}
+
+Handle(Graphic3d_GraphicDriver) Graphic3d::InitGraphicDriver (const Handle(Aspect_DisplayConnection)& theDisplayConnection)
+{
+     return MetaGraphicDriverFactory("");
+}
+#endif
Index: oce/src/NCollection/NCollection_Mat4.hxx
===================================================================
--- oce.orig/src/NCollection/NCollection_Mat4.hxx
+++ oce/src/NCollection/NCollection_Mat4.hxx
@@ -203,8 +203,6 @@ public:
   //! Raw access to the data (for OpenGL exchange).
   const Element_t* GetData()    const { return myMat; }
   Element_t*       ChangeData()       { return myMat; }
-  operator const   Element_t*() const { return myMat; }
-  operator         Element_t*()       { return myMat; }
 
   //! Multiply by the vector (M * V).
   //! @param theVec [in] the vector to multiply.
@@ -228,10 +226,10 @@ public:
     size_t aInputElem;
     for (size_t aResElem = 0; aResElem < 16; ++aResElem)
     {
-      aMatRes[aResElem] = (Element_t )0;
+      aMatRes.myMat[aResElem] = (Element_t )0;
       for (aInputElem = 0; aInputElem < 4; ++aInputElem)
       {
-        aMatRes[aResElem] += theMatA.GetValue(aResElem % 4, aInputElem)
+        aMatRes.myMat[aResElem] += theMatA.GetValue(aResElem % 4, aInputElem)
                            * theMatB.GetValue(aInputElem, aResElem / 4);
       }
     }
Index: oce/src/NCollection/NCollection_Vec3.hxx
===================================================================
--- oce.orig/src/NCollection/NCollection_Vec3.hxx
+++ oce/src/NCollection/NCollection_Vec3.hxx
@@ -161,14 +161,14 @@ public:
   friend NCollection_Vec3 operator+ (const NCollection_Vec3& theLeft,
                                      const NCollection_Vec3& theRight)
   {
-    NCollection_Vec3 aSumm = NCollection_Vec3 (theLeft);
+    NCollection_Vec3 aSumm = NCollection_Vec3<Element_t> (theLeft);
     return aSumm += theRight;
   }
 
   //! Unary -.
   NCollection_Vec3 operator-() const
   {
-    return NCollection_Vec3 (-x(), -y(), -z());
+    return NCollection_Vec3<Element_t> (-x(), -y(), -z());
   }
 
   //! Compute per-component subtraction.
@@ -311,19 +311,19 @@ public:
   //! Constuct DX unit vector.
   static NCollection_Vec3 DX()
   {
-    return NCollection_Vec3 (Element_t(1), Element_t(0), Element_t(0));
+    return NCollection_Vec3<Element_t> (Element_t(1), Element_t(0), Element_t(0));
   }
 
   //! Constuct DY unit vector.
   static NCollection_Vec3 DY()
   {
-    return NCollection_Vec3 (Element_t(0), Element_t(1), Element_t(0));
+    return NCollection_Vec3<Element_t> (Element_t(0), Element_t(1), Element_t(0));
   }
 
   //! Constuct DZ unit vector.
   static NCollection_Vec3 DZ()
   {
-    return NCollection_Vec3 (Element_t(0), Element_t(0), Element_t(1));
+    return NCollection_Vec3<Element_t> (Element_t(0), Element_t(0), Element_t(1));
   }
 
 private:
Index: oce/src/NCollection/NCollection_Vec4.hxx
===================================================================
--- oce.orig/src/NCollection/NCollection_Vec4.hxx
+++ oce/src/NCollection/NCollection_Vec4.hxx
@@ -215,7 +215,7 @@ public:
   //! Unary -.
   NCollection_Vec4 operator-() const
   {
-    return NCollection_Vec4 (-x(), -y(), -z(), -w());
+    return NCollection_Vec4<Element_t> (-x(), -y(), -z(), -w());
   }
 
   //! Compute per-component subtraction.
@@ -232,7 +232,7 @@ public:
   friend NCollection_Vec4 operator- (const NCollection_Vec4& theLeft,
                                      const NCollection_Vec4& theRight)
   {
-    NCollection_Vec4 aSumm = NCollection_Vec4 (theLeft);
+    NCollection_Vec4 aSumm = NCollection_Vec4<Element_t> (theLeft);
     return aSumm -= theRight;
   }
 
Index: oce/src/OSD/OSD_FileNode.cxx
===================================================================
--- oce.orig/src/OSD/OSD_FileNode.cxx
+++ oce/src/OSD/OSD_FileNode.cxx
@@ -945,16 +945,13 @@ void _osd_wnt_set_error ( OSD_Error& err
 
  DWORD              errCode;
  Standard_Character buffer[ 2048 ];
- va_list            arg_ptr;
-
- va_start ( arg_ptr, err );
 
  errCode = GetLastError ();
 
  if (  !FormatMessage (
-         FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_ARGUMENT_ARRAY,
+         FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
          0, errCode, MAKELANGID( LANG_NEUTRAL, SUBLANG_NEUTRAL ),
-         buffer, 2048, &arg_ptr
+         buffer, 2048, NULL
         )
  ) {
  
@@ -965,8 +962,6 @@ void _osd_wnt_set_error ( OSD_Error& err
 
  err.SetValue ( errCode, who, buffer );
 
- va_end ( arg_ptr );
-
 }  // end _set_error
 
 
Index: oce/src/OpenGl/OpenGl_Context.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_Context.cxx
+++ oce/src/OpenGl/OpenGl_Context.cxx
@@ -75,7 +75,6 @@ OpenGl_Context::OpenGl_Context (const Ha
   core14 (NULL),
   core15 (NULL),
   core20 (NULL),
-  caps   (!theCaps.IsNull() ? theCaps : new OpenGl_Caps()),
   arbNPTW(Standard_False),
   arbVBO (NULL),
   arbTBO (NULL),
@@ -102,6 +101,11 @@ OpenGl_Context::OpenGl_Context (const Ha
   myIsFeedback (Standard_False),
   myIsInitialized (Standard_False)
 {
+  if (!theCaps.IsNull())
+    caps = theCaps;
+  else
+    caps = new OpenGl_Caps();
+
 #if defined(MAC_OS_X_VERSION_10_3) && !defined(MACOSX_USE_GLX)
   // Vendors can not extend functionality on this system
   // and developers are limited to OpenGL support provided by Mac OS X SDK.
@@ -592,7 +596,7 @@ static void APIENTRY debugCallbackWrap(u
                                        unsigned int theSeverity,
                                        int          /*theLength*/,
                                        const char*  theMessage,
-                                       void*  theUserParam)
+                                       const void*  theUserParam)
 {
   OpenGl_Context* aCtx = (OpenGl_Context* )theUserParam;
   aCtx->PushMessage (theSource, theType, theId, theSeverity, theMessage);
Index: oce/src/OpenGl/OpenGl_Display.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_Display.cxx
+++ oce/src/OpenGl/OpenGl_Display.cxx
@@ -35,7 +35,7 @@ IMPLEMENT_STANDARD_RTTIEXT(OpenGl_Displa
 namespace
 {
   #if (defined(_WIN32) || defined(__WIN32__)) || (defined(__APPLE__) && !defined(MACOSX_USE_GLX))
-    static char* TheDummyDisplay = "DISPLAY";
+    static char TheDummyDisplay[] = "DISPLAY";
   #endif
 
   static const OpenGl_Facilities myDefaultFacilities = { 1, 1, 1, 1, OpenGLMaxLights, 10000 };
Index: oce/src/OpenGl/OpenGl_Flipper.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_Flipper.cxx
+++ oce/src/OpenGl/OpenGl_Flipper.cxx
@@ -192,7 +192,7 @@ void OpenGl_Flipper::Render (const Handl
     glMatrixMode (GL_MODELVIEW);
   }
 
-  glLoadMatrixf ((GLfloat*) aMatrixMV);
+  glLoadMatrixf (aMatrixMV.GetData());
 
   if (aCurrMode != GL_MODELVIEW)
   {
Index: oce/src/OpenGl/OpenGl_GlCore12.hxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_GlCore12.hxx
+++ oce/src/OpenGl/OpenGl_GlCore12.hxx
@@ -20,13 +20,15 @@
 
 #include <OpenGl_GlCore11.hxx>
 
-#if defined(__APPLE__) && !defined(MACOSX_USE_GLX)
-  #undef GL_VERSION_1_2
-  #undef GL_VERSION_1_3
-  #undef GL_VERSION_1_4
-  #undef GL_VERSION_1_5
-  #undef GL_VERSION_2_0
-#endif
+// We can safely #undef GL_VERSION_x_y since redeclaration of typedef names can
+// be done for same type: http://msdn.microsoft.com/en-us/library/87txds41.aspx
+// It is required to fix build against Mesa >= 10:
+// http://cgit.freedesktop.org/mesa/mesa/commit/?id=a36f7e6
+#undef GL_VERSION_1_2
+#undef GL_VERSION_1_3
+#undef GL_VERSION_1_4
+#undef GL_VERSION_1_5
+#undef GL_VERSION_2_0
 
 #include <OpenGl_glext.h>
 
Index: oce/src/OpenGl/OpenGl_ShaderManager.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_ShaderManager.cxx
+++ oce/src/OpenGl/OpenGl_ShaderManager.cxx
@@ -611,7 +611,7 @@ static void PushAspectFace (const Handle
 {
   theProgram->SetUniform (theCtx,
                           theProgram->GetStateLocation (OpenGl_OCCT_TEXTURE_ENABLE),
-                          theAspect->DoTextureMap());
+                          (GLint)theAspect->DoTextureMap());
   theProgram->SetUniform (theCtx,
                           theProgram->GetStateLocation (OpenGl_OCCT_ACTIVE_SAMPLER),
                           0 /* GL_TEXTURE0 */);
Index: oce/src/QABugs/QABugs_3.cxx
===================================================================
--- oce.orig/src/QABugs/QABugs_3.cxx
+++ oce/src/QABugs/QABugs_3.cxx
@@ -1012,7 +1012,10 @@ static Standard_Integer coordload (Draw_
   }
 
   BRepBuilderAPI_MakeFace aMakeFace (aMakeWire.Wire());
-  DBRep::Set (theArgVec[1], aMakeFace.IsDone() ? aMakeFace.Face() : TopoDS_Face());
+  TopoDS_Face aFace;
+  if (aMakeFace.IsDone())
+    aFace = aMakeFace.Face();
+  DBRep::Set (theArgVec[1], aFace);
   return 0;
 }
 
Index: oce/src/Standard/Standard_Atomic.hxx
===================================================================
--- oce.orig/src/Standard/Standard_Atomic.hxx
+++ oce/src/Standard/Standard_Atomic.hxx
@@ -27,6 +27,8 @@
 #ifndef _Standard_Atomic_HeaderFile
 #define _Standard_Atomic_HeaderFile
 
+#include <Standard_Macro.hxx>
+
 //! Increments atomically integer variable pointed by theValue
 //! and returns resulting incremented value.
 inline int Standard_Atomic_Increment (volatile int* theValue);
Index: oce/src/Standard/Standard_TypeDef.hxx
===================================================================
--- oce.orig/src/Standard/Standard_TypeDef.hxx
+++ oce/src/Standard/Standard_TypeDef.hxx
@@ -18,19 +18,16 @@
 #include <cstddef>
 #include <ctime>
 
-#if(defined(_MSC_VER) && (_MSC_VER < 1600))
+#if(defined(_MSC_VER) && (_MSC_VER < 1800)) || defined(__BORLANDC__)
+
+  #if defined(_MSC_VER) && (_MSC_VER < 1600)
   // old MSVC - hasn't stdint header
   typedef unsigned __int16  uint16_t;
   typedef unsigned __int32  uint32_t;
-#else
-#if defined(__hpux) && !defined(__GNUC__)
-  #include <inttypes.h>
-#else
+  #else
   #include <stdint.h>
-#endif
-#endif
+  #endif
 
-#if(defined(_MSC_VER) && (_MSC_VER < 1800))
   // only Visual Studio 2013 (vc12) provides <cinttypes> header
   // we do not defined all macros here - only used by OCCT framework
   #ifdef _WIN64
Index: oce/src/TObj/TObj_Model.cxx
===================================================================
--- oce.orig/src/TObj/TObj_Model.cxx
+++ oce/src/TObj/TObj_Model.cxx
@@ -232,7 +232,7 @@ Handle(TCollection_HAsciiString) TObj_Mo
   TCollection_AsciiString aPath (aDoc->GetPath());
   return !aPath.IsEmpty()
        ? new TCollection_HAsciiString (aPath)
-       : Handle(TCollection_HAsciiString)();
+       : new TCollection_HAsciiString ();
 }
 
 //=======================================================================
Index: oce/src/ViewerTest/ViewerTest_CMPLRS.edl
===================================================================
--- oce.orig/src/ViewerTest/ViewerTest_CMPLRS.edl
+++ oce/src/ViewerTest/ViewerTest_CMPLRS.edl
@@ -17,6 +17,6 @@
 
   @uses "CSF.edl" ;
 
-  @string %CMPLRS_CXX_Options += " " %CSF_TCL_INCLUDE;
+  @string %CMPLRS_CXX_Options += " " %CSF_TCL_INCLUDE " " %CSF_FREETYPE_INCLUDES " ";
 
 @endif;
Index: oce/inc/BOPDS_Curve.hxx
===================================================================
--- oce.orig/inc/BOPDS_Curve.hxx
+++ oce/inc/BOPDS_Curve.hxx
@@ -34,6 +34,9 @@
 #ifndef _Handle_BOPDS_PaveBlock_HeaderFile
 #include <Handle_BOPDS_PaveBlock.hxx>
 #endif
+#ifndef _BOPDS_PaveBlock_HeaderFile
+#include <BOPDS_PaveBlock.hxx>
+#endif
 #ifndef _Standard_Boolean_HeaderFile
 #include <Standard_Boolean.hxx>
 #endif
Index: oce/inc/IntPatch_PrmPrmIntersection.hxx
===================================================================
--- oce.orig/inc/IntPatch_PrmPrmIntersection.hxx
+++ oce/inc/IntPatch_PrmPrmIntersection.hxx
@@ -37,6 +37,9 @@
 #ifndef _Handle_IntPatch_Line_HeaderFile
 #include <Handle_IntPatch_Line.hxx>
 #endif
+#ifndef _IntPatch_Polyhedron_HeaderFile
+#include <IntPatch_Polyhedron.hxx>
+#endif
 #ifndef _Handle_IntSurf_LineOn2S_HeaderFile
 #include <Handle_IntSurf_LineOn2S.hxx>
 #endif
Index: oce/src/StdPrs/StdPrs_HLRShape.hxx
===================================================================
--- oce.orig/src/StdPrs/StdPrs_HLRShape.hxx
+++ oce/src/StdPrs/StdPrs_HLRShape.hxx
@@ -18,6 +18,7 @@
 #include <Handle_Prs3d_Presentation.hxx>
 #include <Handle_Prs3d_Drawer.hxx>
 #include <Handle_Prs3d_Projector.hxx>
+#include <TopoDS_Shape.hxx>
 
 class StdPrs_HLRShape: public Prs3d_Root
 {
Index: oce/inc/config.h
===================================================================
--- oce.orig/inc/config.h
+++ oce/inc/config.h
@@ -938,9 +938,6 @@
 /* Define to 1 if you have the <X11/extensions/transovl.h> header file. */
 /* #undef HAVE_X11_EXTENSIONS_TRANSOVL_H */
 
-/* Define to 1 if you have the <Xmu/Editres.h> header file. */
-/* #undef HAVE_XMU_EDITRES_H */
-
 /* Define to 1 if your C compiler doesn't accept -c and -o together. */
 /* #undef NO_MINUS_C_MINUS_O */
 
@@ -1286,9 +1283,6 @@
 /* Define to 1 if you have the <X11/extensions/transovl.h> header file. */
 /* #undef HAVE_X11_EXTENSIONS_TRANSOVL_H */
 
-/* Define to 1 if you have the <Xmu/Editres.h> header file. */
-/* #undef HAVE_XMU_EDITRES_H */
-
 /* Define to 1 if your C compiler doesn't accept -c and -o together. */
 /* #undef NO_MINUS_C_MINUS_O */
 
@@ -1785,9 +1779,6 @@
 /* Define to 1 if you have the <X11/extensions/transovl.h> header file. */
 /* #undef HAVE_X11_EXTENSIONS_TRANSOVL_H */
 
-/* Define to 1 if you have the <Xmu/Editres.h> header file. */
-#define HAVE_XMU_EDITRES_H 1
-
 /* Define to 1 if your C compiler doesn't accept -c and -o together. */
 /* #undef NO_MINUS_C_MINUS_O */
 
Index: oce/scripts/gendoc
===================================================================
--- oce.orig/scripts/gendoc
+++ /dev/null
@@ -1,19 +0,0 @@
-#!/bin/bash
-
-# Helper script to run generation of OCCT documentation on Linux.
-# Running it requires that Tcl, Doxygen, and MikTex (for PDF generation) should be in the PATH
-
-anArgs=$*
-anOldPath="$PATH"
-anOldLd="$LD_LIBRARY_PATH"
-anOldDyLd="$DYLD_LIBRARY_PATH"
-
-# go to the script directory
-aScriptPath=${BASH_SOURCE%/*}; if [ -d "${aScriptPath}" ]; then cd "$aScriptPath"; fi; aScriptPath="$PWD";
-if [ -e "${aScriptPath}/env.sh" ]; then source "${aScriptPath}/env.sh"; fi
-
-tclsh "${aScriptPath}/adm/start.tcl" $anArgs
-
-export PATH="$anOldPath"
-export LD_LIBRARY_PATH="$anOldLd"
-export DYLD_LIBRARY_PATH="$anOldDyLd"
Index: oce/src/FSD/FSD_FileHeader.hxx
===================================================================
--- oce.orig/src/FSD/FSD_FileHeader.hxx
+++ oce/src/FSD/FSD_FileHeader.hxx
@@ -33,12 +33,21 @@ struct FSD_FileHeader {
 };
 
 #ifndef DO_INVERSE
+# ifdef HAVE_CONFIG_H
+#  include <oce-config.h>
+#  ifdef WORDS_BIGENDIAN
+#   define DO_INVERSE 1
+#  else
+#   define DO_INVERSE 0
+#  endif
+# else
 #if defined ( SOLARIS ) || defined ( IRIX ) || defined(__hpux)
 #define DO_INVERSE 1
 #else
 #define DO_INVERSE 0
 #endif
 #endif
+#endif
 
 //=======================================================================
 //function : InverseInt
Index: oce/src/Intf/Intf_InterferencePolyhedron.gxx
===================================================================
--- oce.orig/src/Intf/Intf_InterferencePolyhedron.gxx
+++ oce/src/Intf/Intf_InterferencePolyhedron.gxx
@@ -894,7 +894,7 @@ Standard_Boolean Intf_InterferencePolyhe
 	if (nob<0) break;
       }
       if (nob>=0) {
-	while (nob<nbNoInserted) {
+	while ((nob<nbNoInserted) && (nob + 1 < static_cast<Standard_Integer>(sizeof(piToInsert)))) {
 	  piToInsert[nob]=piToInsert[nob+1];
 	  nob++;
 	}
Index: oce/src/LDOM/LDOMBasicString.cxx
===================================================================
--- oce.orig/src/LDOM/LDOMBasicString.cxx
+++ oce/src/LDOM/LDOMBasicString.cxx
@@ -137,8 +137,10 @@ LDOMBasicString& LDOMBasicString::operat
 
 LDOMBasicString& LDOMBasicString::operator = (const LDOMBasicString& anOther)
 {
-  if (myType == LDOM_AsciiFree && myVal.ptr)
+  if (myType == LDOM_AsciiFree && myVal.ptr) {
+    if (myVal.ptr == anOther.myVal.ptr && myType == anOther.Type()) return * this;
     delete [] (char *) myVal.ptr;
+  }
   myType    = anOther.Type();
   switch (myType) {
   case LDOM_AsciiFree:
