From 36fcfe8f4274870476cecacce927cb19fa2d6ce4 Mon Sep 17 00:00:00 2001
From: tpaviot <tpaviot@gmail.com>
Date: Thu, 28 Jul 2011 09:12:06 +0200
Subject: [PATCH 3/9] [cppcheck-fix][resource-leak]

---
 src/DrawResources/DIFF.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

Index: oce/src/DrawResources/DIFF.c
===================================================================
--- oce.orig/src/DrawResources/DIFF.c
+++ oce/src/DrawResources/DIFF.c
@@ -17,6 +17,7 @@
   FILE *fp2 = fopen(argv[2],"r");
   if(fp2 == NULL) { 
     fprintf(stderr,"\n Unable to open file2:%s \n",argv[2]);
+    fclose(fp1);
     return(-3);
   }
 
@@ -38,7 +39,8 @@
       }    
     }
   }
-  
+  fclose(fp1);
+  fclose(fp2);
   return(0);
 }
   
