From c784b1682bc43ef9485850d26b5e2d57d9bb7825 Mon Sep 17 00:00:00 2001
From: tpaviot <tpaviot@gmail.com>
Date: Thu, 28 Jul 2011 08:32:19 +0200
Subject: [PATCH 2/9] [cppcheck-fix][uninitialized-var]

---
 inc/ApproxInt_Approx.gxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: oce/inc/ApproxInt_Approx.gxx
===================================================================
--- oce.orig/inc/ApproxInt_Approx.gxx
+++ oce/inc/ApproxInt_Approx.gxx
@@ -99,7 +99,7 @@
     pfunc = &IntSurf_PntOn2S::ParametersOnS2;
   for(Standard_Integer i=1;i<=nbp;i++) { 
     const IntSurf_PntOn2S&  POn2S = theline->Point(i);
-    Standard_Real  U,V;
+    Standard_Real  U=0.0,V=0.0;
     (POn2S.*pfunc)(U,V);
     if(U<u0) u0=U;
     if(U>u1) u1=U;
