commit 6795f0fd525e33751b7cae77ff18532616dbb89c
Author: QbProg <tholag@gmail.com>
Date:   Sun Jun 19 18:20:09 2011 +0200

    Marked @todos related to some of the MSVC warnings

Index: oce/inc/Prs2d_Dimension.hxx
===================================================================
--- oce.orig/inc/Prs2d_Dimension.hxx
+++ oce/inc/Prs2d_Dimension.hxx
@@ -122,7 +122,8 @@
         Standard_Boolean ArrowIsReversed() const;
   //! Return true if symbol is drawn, false - in otherwise <br>
         Standard_Boolean IsDrawSymbol() const;
-  //! Return ASCII code of the symbol <br>
+  //! Return ASCII code of the symbol <br> 
+  // @todo Should really be Standard_Character
         Standard_Integer SymbolCode() const;
   
   Standard_EXPORT    const Handle_TColgp_HArray1OfPnt2d ArrayOfFirstArrowPnt() const;
Index: oce/src/Aspect/Aspect.cxx
===================================================================
--- oce.orig/src/Aspect/Aspect.cxx
+++ oce/src/Aspect/Aspect.cxx
@@ -100,7 +100,7 @@
          estring.SetValue(i,c);
       } else if( (c > Standard_ExtCharacter(0x0020)) && 
 			(c <= Standard_ExtCharacter(0x007F)) ) {
-	c += Standard_ExtCharacter(0xFEE0);
+	c += Standard_ExtCharacter(0xFEE0); // @todo Possible bug here : cast truncates constant value 
 	estring.SetValue(i,c);
       }
     }
Index: oce/src/DDF/DDF_IOStream.cxx
===================================================================
--- oce.orig/src/DDF/DDF_IOStream.cxx
+++ oce/src/DDF/DDF_IOStream.cxx
@@ -296,6 +296,13 @@
 
   IsEnd = Standard_False;
 
+  /* @todo c is used before initialized. Anyway I didn't modify the code because
+   I don't know if ReadString should include \n in the output.
+   If so : Swap buffer += c with the next line
+   If not: rewrite the loop 
+   I don't understand anyway the semantics of this function.
+   QbProg
+  */
   while (!IsEnd && !DDF_IOStream::IsEnd()) {
     buffer += c;
     myIStream->get(c);
@@ -321,7 +328,14 @@
   }
 
   IsEnd = Standard_False;
-
+  
+  /* @todo c is used before initialized. Anyway I didn't modify the code because
+   I don't know if ReadWord should include \n in the output.
+   If so : Swap buffer += c with the next line
+   If not: rewrite the loop 
+   I don't understand anyway the semantics of this function.
+   QbProg
+  */
   while (!IsEnd && !DDF_IOStream::IsEnd()) {
     buffer += c;
     myIStream->get(c);
Index: oce/src/IFSelect/IFSelect_WorkSession.cxx
===================================================================
--- oce.orig/src/IFSelect/IFSelect_WorkSession.cxx
+++ oce/src/IFSelect/IFSelect_WorkSession.cxx
@@ -2814,7 +2814,7 @@
 {
   if (!IsLoaded()) return;
   Standard_Integer i,nb = themodel->NbEntities();
-  thecheckana = TCollection_AsciiString (' ',nb+1);
+  thecheckana = TCollection_AsciiString (' ',nb+1); // @todo Probably the constructor here has to be called with inversed parameters (nb+1,' ')
   for (chl.Start(); chl.More(); chl.Next()) {
     Standard_Integer num = chl.Number();
     const Handle(Interface_Check) ach = chl.Value();
Index: oce/src/Visual3d/Visual3d_ViewManager.cxx
===================================================================
--- oce.orig/src/Visual3d/Visual3d_ViewManager.cxx
+++ oce/src/Visual3d/Visual3d_ViewManager.cxx
@@ -985,7 +985,7 @@
 // The marking is activated only if the data is correct
 Standard_Boolean DoPick = Standard_False;
 
-CALL_DEF_PICK apick;
+CALL_DEF_PICK apick; //@todo CALL_DEF_PICK should have constructor
 Standard_Integer Width, Height;
 
 	// Parse the list of views to find a 
