diff --git a/test/BRepAlgoAPI_test/CMakeLists.txt b/test/BRepAlgoAPI_test/CMakeLists.txt
index e60a92d..05379ac 100644
--- a/test/BRepAlgoAPI_test/CMakeLists.txt
+++ b/test/BRepAlgoAPI_test/CMakeLists.txt
@@ -1,8 +1 @@
-FILE(GLOB TEST_SOURCES *.cpp)
-SET(SOURCE_FILES ${TEST_SOURCES})
-
-ADD_EXECUTABLE(boolean_ops_test ${SOURCE_FILES})
-TARGET_LINK_LIBRARIES(boolean_ops_test TKPrim TKBO)
-TARGET_LINK_LIBRARIES(boolean_ops_test gcov ${GTEST_LIBRARY} ${GTEST_MAIN_LIBRARY})
-
-GTEST_ADD_TESTS(boolean_ops_test "" ${SOURCE_FILES})
+ADD_OCE_TEST(boolean_ops_test "TKPrim;TKBO")
diff --git a/test/BRepExtrema_test/BRepExtrema_test.cpp b/test/BRepExtrema_test/BRepExtrema_test.cpp
new file mode 100644
index 0000000..f24c512
--- /dev/null
+++ b/test/BRepExtrema_test/BRepExtrema_test.cpp
@@ -0,0 +1,81 @@
+#include <BRepExtrema_DistShapeShape.hxx>
+#include <BRepPrimAPI_MakeSphere.hxx>
+#include <TopoDS_Shape.hxx>
+#include <gp_Pnt.hxx>
+#include <gp_Vec.hxx>
+#include <BRepBuilderAPI_MakeVertex.hxx>
+
+#include <gtest/gtest.h>
+
+/* @TODO: in the following tests, minimal distances between two shapes.
+It's not obvious anyway to figure out the relationship between the deflection passed to
+BRepExtrema_DistSS (related to the mesh quality) and the precision of the output.
+This has to be investigated further.
+*/
+
+TEST(BRepExtremaTestSuite, testDistSphereSphere)
+{
+    // the distance d between sphere1, centered at origin and radius r1, and
+    // sphere 2, centered at (a,0,0) and radius r2
+    // should be d = a-(r1+r2)
+    Standard_Real r1 = 10.0;
+    BRepPrimAPI_MakeSphere sphere1(r1);
+    TopoDS_Shape sphere1_shp = sphere1.Shape();
+     
+    Standard_Real a = 100.0;
+    Standard_Real r2 = 20.0;
+    gp_Pnt center2(a,0,0);
+    BRepPrimAPI_MakeSphere sphere2(center2,r2);
+    TopoDS_Shape sphere2_shp = sphere2.Shape();
+    
+    BRepExtrema_DistShapeShape distSS;
+    distSS.LoadS1(sphere1_shp);
+    distSS.LoadS2(sphere2_shp);
+    distSS.Perform();
+    
+    // should only be one minimal value
+    ASSERT_EQ(distSS.NbSolution(),1);
+    // this value should be
+    Standard_Real expected_result = a-(r1+r2);
+    Standard_Real min_dist = distSS.Value();
+    // @TODO: should compare with a precision
+    float abs_error = 0.001;
+    ASSERT_NEAR(expected_result,min_dist,abs_error);
+}
+
+TEST(BRepExtremaTestSuite, testDistSphereVertex)
+{
+    // the distance d between sphere1, centered at origin and radius r1, and
+    // vertex, a point of coordinates (a,b,c)
+    // should be d = sqrt(a^2+b^2+c^2)-r1
+    Standard_Real r1 = 10.0;
+    BRepPrimAPI_MakeSphere sphere1(r1);
+    TopoDS_Shape sphere1_shp = sphere1.Shape();
+     
+    Standard_Real a = 100.0;
+    Standard_Real b = 50.0;
+    Standard_Real c = 20.0;
+    
+    gp_Pnt point(a,b,c);
+    TopoDS_Vertex V=BRepBuilderAPI_MakeVertex(point).Vertex();
+    BRepExtrema_DistShapeShape distSS;
+    distSS.LoadS1(sphere1_shp);
+    distSS.LoadS2(V);
+    distSS.SetDeflection(0.001);
+    distSS.Perform();
+    
+    // should only be one minimal value
+    ASSERT_EQ(distSS.NbSolution(),1);
+    // this value should be
+    gp_Vec v(a,b,c);
+    Standard_Real expected_result = v.Magnitude()-r1;
+    Standard_Real min_dist = distSS.Value();
+    
+    float abs_error = 0.001;
+    ASSERT_NEAR(expected_result,min_dist,abs_error);
+}
+
+int main(int argc, char **argv){
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
diff --git a/test/BRepExtrema_test/CMakeLists.txt b/test/BRepExtrema_test/CMakeLists.txt
new file mode 100644
index 0000000..808eaff
--- /dev/null
+++ b/test/BRepExtrema_test/CMakeLists.txt
@@ -0,0 +1 @@
+ADD_OCE_TEST(BRepExtrema_test "TKPrim")
diff --git a/test/BRepGProp_test/BRepGProp_test.cpp b/test/BRepGProp_test/BRepGProp_test.cpp
index 9368f30..dc50898 100644
--- a/test/BRepGProp_test/BRepGProp_test.cpp
+++ b/test/BRepGProp_test/BRepGProp_test.cpp
@@ -1,5 +1,5 @@
 #include <BRepPrimAPI_MakeBox.hxx>
-#include <BRepPrimAPI_MakeSPhere.hxx>
+#include <BRepPrimAPI_MakeSphere.hxx>
 #include <BRepGProp.hxx>
 #include <GProp_GProps.hxx>
 #include <gtest/gtest.h>
diff --git a/test/BRepGProp_test/CMakeLists.txt b/test/BRepGProp_test/CMakeLists.txt
index ab6c288..23d2614 100644
--- a/test/BRepGProp_test/CMakeLists.txt
+++ b/test/BRepGProp_test/CMakeLists.txt
@@ -1,8 +1 @@
-FILE(GLOB TEST_SOURCES *.cpp)
-SET(SOURCE_FILES ${TEST_SOURCES})
-
-ADD_EXECUTABLE(BRepGProp_test ${SOURCE_FILES})
-TARGET_LINK_LIBRARIES(BRepGProp_test TKG2d TKPrim TKTopAlgo)
-TARGET_LINK_LIBRARIES(BRepGProp_test gcov ${GTEST_LIBRARY} ${GTEST_MAIN_LIBRARY})
-
-GTEST_ADD_TESTS(BRepGProp_test "" ${SOURCE_FILES})
+ADD_OCE_TEST(BRepGProp_test "TKG2d;TKPrim;TKTopAlgo")
diff --git a/test/BRepOffsetAPI_test/BRepOffsetAPI_test.cpp b/test/BRepOffsetAPI_test/BRepOffsetAPI_test.cpp
new file mode 100644
index 0000000..c7267a6
--- /dev/null
+++ b/test/BRepOffsetAPI_test/BRepOffsetAPI_test.cpp
@@ -0,0 +1,36 @@
+#include <BRepBuilderAPI_MakePolygon.hxx>
+#include <BRepPrimAPI_MakeBox.hxx>
+#include <BRepOffsetAPI_MakeEvolved.hxx>
+#include <BRepOffsetAPI_MakeOffsetShape.hxx>
+#include <gp_Pnt.hxx>
+#include <GeomAbs_JoinType.hxx>
+#include <TopoDS_Shape.hxx>
+#include <gtest/gtest.h>
+
+TEST(BRepOffsetAPITestSuite, testEvolvedShape)
+{
+    BRepBuilderAPI_MakePolygon P;
+    P.Add(gp_Pnt(0.,0.,0.));
+    P.Add(gp_Pnt(200.,0.,0.));
+    P.Add(gp_Pnt(200.,200.,0.));
+    P.Add(gp_Pnt(0.,200.,0.));
+    P.Add(gp_Pnt(0.,0.,0.));
+    BRepBuilderAPI_MakePolygon wprof(gp_Pnt(0.,0.,0.),gp_Pnt(-60.,-60.,-200.));
+    BRepOffsetAPI_MakeEvolved S(P.Wire(),wprof.Wire(),GeomAbs_Arc,Standard_True,Standard_False,Standard_True,0.0001);
+    S.Build();
+    ASSERT_TRUE(S.IsDone());
+}
+
+TEST(BRepOffsetAPITestSuite, testOffsetBox)
+{
+    BRepPrimAPI_MakeBox b(gp_Pnt(-100,-60,-80),150,200,170);
+    b.Build();
+    TopoDS_Shape b_shape = b.Shape();
+    BRepOffsetAPI_MakeOffsetShape offsetA(b_shape,60,0.01);
+    ASSERT_TRUE(offsetA.IsDone());
+}
+
+int main(int argc, char **argv){
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
diff --git a/test/BRepOffsetAPI_test/CMakeLists.txt b/test/BRepOffsetAPI_test/CMakeLists.txt
new file mode 100644
index 0000000..c797ccf
--- /dev/null
+++ b/test/BRepOffsetAPI_test/CMakeLists.txt
@@ -0,0 +1 @@
+ADD_OCE_TEST(BRepOffsetAPI_test "TKOffset")
diff --git a/test/BRepPrimAPI_test/CMakeLists.txt b/test/BRepPrimAPI_test/CMakeLists.txt
index 0409132..27d6bcf 100644
--- a/test/BRepPrimAPI_test/CMakeLists.txt
+++ b/test/BRepPrimAPI_test/CMakeLists.txt
@@ -1,8 +1 @@
-FILE(GLOB TEST_SOURCES *.cpp)
-SET(SOURCE_FILES ${TEST_SOURCES})
-
-ADD_EXECUTABLE(BRepPrimAPI_test ${SOURCE_FILES})
-TARGET_LINK_LIBRARIES(BRepPrimAPI_test TKPrim)
-TARGET_LINK_LIBRARIES(BRepPrimAPI_test gcov ${GTEST_LIBRARY} ${GTEST_MAIN_LIBRARY})
-
-GTEST_ADD_TESTS(BRepPrimAPI_test "" ${SOURCE_FILES})
\ No newline at end of file
+ADD_OCE_TEST(BRepPrimAPI_test "TKPrim")
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 8831e8b..ea6145b 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -2,8 +2,53 @@
 # The OCE unittest framework use GoogleTest 1.6.0 (aka GTest)
 # The GTest source code is available in the /test/gtest-1.6.0 folder
 #
-SET(GTEST_ROOT ./gtest-1.6.0)
-FIND_PACKAGE(GTest)
+SET(GTEST_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/gtest-1.6.0)
+
+INCLUDE_DIRECTORIES(${GTEST_ROOT}/include ${GTEST_ROOT})
+ADD_LIBRARY(gtest STATIC ${GTEST_ROOT}/src/gtest-all.cc )
+TARGET_LINK_LIBRARIES(gtest ${CMAKE_THREAD_LIBS_INIT})
+
+IF (USE_SOLUTION_FOLDERS)
+	SET_TARGET_PROPERTIES(gtest PROPERTIES FOLDER "Tests" )
+ENDIF(USE_SOLUTION_FOLDERS)
+
+# Copied from CMake
+# Thanks to Daniel Blezek <blezek@gmail.com> for the GTEST_ADD_TESTS code
+function(GTEST_ADD_TESTS executable extra_args)
+    if(NOT ARGN)
+        message(FATAL_ERROR "Missing ARGN: Read the documentation for GTEST_ADD_TESTS")
+    endif()
+    foreach(source ${ARGN})
+        file(READ "${source}" contents)
+        string(REGEX MATCHALL "TEST_?F?\\(([A-Za-z_0-9 ,]+)\\)" found_tests ${contents})
+        foreach(hit ${found_tests})
+            string(REGEX REPLACE ".*\\( *([A-Za-z_0-9]+), *([A-Za-z_0-9]+) *\\).*" "\\1.\\2" test_name ${hit})
+            add_test(NAME ${test_name} COMMAND ${executable} --gtest_filter=${test_name} ${extra_args})
+        endforeach()
+    endforeach()
+endfunction()
+
+MACRO(ADD_OCE_TEST test_name libs)
+  FILE(GLOB TEST_SOURCES *.cpp)
+  SET(SOURCE_FILES ${TEST_SOURCES})
+  ADD_EXECUTABLE(${test_name} ${SOURCE_FILES})
+ 
+  IF (USE_SOLUTION_FOLDERS)
+		SET_TARGET_PROPERTIES(${test_name} PROPERTIES FOLDER "Tests" )
+  ENDIF(USE_SOLUTION_FOLDERS)
+  
+  FOREACH(LIB ${libs})
+     TARGET_LINK_LIBRARIES(${test_name} ${LIB})
+  ENDFOREACH(LIB ${libs})
+  
+  IF (${PROJECT_NAME}_COVERAGE)
+    TARGET_LINK_LIBRARIES(${test_name} gcov gtest)
+  ELSE (${PROJECT_NAME}_COVERAGE)
+    TARGET_LINK_LIBRARIES(${test_name} gtest)
+  ENDIF (${PROJECT_NAME}_COVERAGE)
+  GTEST_ADD_TESTS(${test_name} "" ${SOURCE_FILES})
+ENDMACRO(ADD_OCE_TEST test_name libs)
 
 FILE(GLOB TEST_DIRECTORIES ./*_test gtest_sandbox)
 SUBDIRS(${TEST_DIRECTORIES})
+
diff --git a/test/gp_test/CMakeLists.txt b/test/gp_test/CMakeLists.txt
index 200931c..1ae4675 100644
--- a/test/gp_test/CMakeLists.txt
+++ b/test/gp_test/CMakeLists.txt
@@ -1,8 +1 @@
-FILE(GLOB TEST_SOURCES *.cpp)
-SET(SOURCE_FILES ${TEST_SOURCES})
-
-ADD_EXECUTABLE(gp_test ${SOURCE_FILES})
-TARGET_LINK_LIBRARIES(gp_test TKG2d)
-TARGET_LINK_LIBRARIES(gp_test gcov ${GTEST_LIBRARY} ${GTEST_MAIN_LIBRARY})
-
-GTEST_ADD_TESTS(gp_test "" ${SOURCE_FILES})
+ADD_OCE_TEST(gp_test "TKG2d")
diff --git a/test/gp_test/gp_test.cpp b/test/gp_test/gp_test.cpp
index 1187f6c..bcae9a1 100644
--- a/test/gp_test/gp_test.cpp
+++ b/test/gp_test/gp_test.cpp
@@ -24,6 +24,20 @@ TEST(gpTestSuite, testBuildThousandsPoints)
     }
 }
 
+/*  This test fails with OCCT 6.5.1 and GCC 4.2 */
+TEST(gpTestSuite, testGCC42gp_Trsf)
+{
+    gp_Trsf T;
+    T.SetValues(
+        1.0, 0.0, 0.0, 0.0,
+        0.0, 1.0, 0.0, 0.0,
+        0.0, 0.0, 1.0, 0.0,
+        1.e-6, 1.e-6);
+    ASSERT_EQ(1.0,T.Value(1,1));
+    ASSERT_EQ(1.0,T.Value(2,2));
+    ASSERT_EQ(1.0,T.Value(3,3));
+}
+
 int main(int argc, char **argv){
   testing::InitGoogleTest(&argc, argv);
   return RUN_ALL_TESTS();
diff --git a/test/gtest-1.6.0/build-aux/config.guess b/test/gtest-1.6.0/build-aux/config.guess
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/build-aux/config.sub b/test/gtest-1.6.0/build-aux/config.sub
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/build-aux/depcomp b/test/gtest-1.6.0/build-aux/depcomp
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/build-aux/install-sh b/test/gtest-1.6.0/build-aux/install-sh
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/build-aux/ltmain.sh b/test/gtest-1.6.0/build-aux/ltmain.sh
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/build-aux/missing b/test/gtest-1.6.0/build-aux/missing
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/configure b/test/gtest-1.6.0/configure
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/msvc/gtest-md.sln b/test/gtest-1.6.0/msvc/gtest-md.sln
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/msvc/gtest-md.vcproj b/test/gtest-1.6.0/msvc/gtest-md.vcproj
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/msvc/gtest.sln b/test/gtest-1.6.0/msvc/gtest.sln
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/msvc/gtest.vcproj b/test/gtest-1.6.0/msvc/gtest.vcproj
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/msvc/gtest_main-md.vcproj b/test/gtest-1.6.0/msvc/gtest_main-md.vcproj
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/msvc/gtest_main.vcproj b/test/gtest-1.6.0/msvc/gtest_main.vcproj
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/msvc/gtest_prod_test-md.vcproj b/test/gtest-1.6.0/msvc/gtest_prod_test-md.vcproj
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/msvc/gtest_prod_test.vcproj b/test/gtest-1.6.0/msvc/gtest_prod_test.vcproj
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/msvc/gtest_unittest-md.vcproj b/test/gtest-1.6.0/msvc/gtest_unittest-md.vcproj
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/msvc/gtest_unittest.vcproj b/test/gtest-1.6.0/msvc/gtest_unittest.vcproj
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/scripts/fuse_gtest_files.py b/test/gtest-1.6.0/scripts/fuse_gtest_files.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/scripts/gen_gtest_pred_impl.py b/test/gtest-1.6.0/scripts/gen_gtest_pred_impl.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/scripts/gtest-config.in b/test/gtest-1.6.0/scripts/gtest-config.in
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/scripts/pump.py b/test/gtest-1.6.0/scripts/pump.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_break_on_failure_unittest.py b/test/gtest-1.6.0/test/gtest_break_on_failure_unittest.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_catch_exceptions_test.py b/test/gtest-1.6.0/test/gtest_catch_exceptions_test.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_color_test.py b/test/gtest-1.6.0/test/gtest_color_test.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_env_var_test.py b/test/gtest-1.6.0/test/gtest_env_var_test.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_filter_unittest.py b/test/gtest-1.6.0/test/gtest_filter_unittest.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_help_test.py b/test/gtest-1.6.0/test/gtest_help_test.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_list_tests_unittest.py b/test/gtest-1.6.0/test/gtest_list_tests_unittest.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_output_test.py b/test/gtest-1.6.0/test/gtest_output_test.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_shuffle_test.py b/test/gtest-1.6.0/test/gtest_shuffle_test.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_test_utils.py b/test/gtest-1.6.0/test/gtest_test_utils.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_throw_on_failure_test.py b/test/gtest-1.6.0/test/gtest_throw_on_failure_test.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_uninitialized_test.py b/test/gtest-1.6.0/test/gtest_uninitialized_test.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_xml_outfiles_test.py b/test/gtest-1.6.0/test/gtest_xml_outfiles_test.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_xml_output_unittest.py b/test/gtest-1.6.0/test/gtest_xml_output_unittest.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/test/gtest_xml_test_utils.py b/test/gtest-1.6.0/test/gtest_xml_test_utils.py
old mode 100644
new mode 100755
diff --git a/test/gtest-1.6.0/xcode/Scripts/versiongenerate.py b/test/gtest-1.6.0/xcode/Scripts/versiongenerate.py
old mode 100644
new mode 100755
diff --git a/test/gtest_sandbox/CMakeLists.txt b/test/gtest_sandbox/CMakeLists.txt
index fcdee27..7b27884 100644
--- a/test/gtest_sandbox/CMakeLists.txt
+++ b/test/gtest_sandbox/CMakeLists.txt
@@ -1,6 +1 @@
-FILE(GLOB TEST_SOURCES *.cpp)
-SET(SOURCE_FILES ${TEST_SOURCES})
-
-ADD_EXECUTABLE(sandbox ${SOURCE_FILES})
-TARGET_LINK_LIBRARIES(sandbox gcov ${GTEST_LIBRARY} ${GTEST_MAIN_LIBRARY})
-ADD_TEST(gtest_sandbox  sandbox)
\ No newline at end of file
+ADD_OCE_TEST(sandbox "")
