commit 10004d9bca2f1e71ff3ece4d5ae6f685d46a920e
Author: Denis Barbier <bouzim@gmail.com>
Date:   Sun Apr 10 13:08:18 2011 +0200

    Fix compilation with FreeImage and GL2PS support

Index: oce/src/Image/Image_PixMap.cxx
===================================================================
--- oce.orig/src/Image/Image_PixMap.cxx
+++ oce/src/Image/Image_PixMap.cxx
@@ -3,6 +3,10 @@
 // Author     KGV
 // Copyright  OpenCASCADE 2010
 
+#ifdef HAVE_CONFIG_H
+# include <config.h>
+#endif
+
 #ifdef HAVE_FREEIMAGE
   #include <FreeImagePlus.h>
   #include <Image_PixMap.ixx>
Index: oce/src/OpenGl/OpenGl_attri.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_attri.cxx
+++ oce/src/OpenGl/OpenGl_attri.cxx
@@ -98,6 +98,10 @@ used in immediat mode context.
 * Includes
 */
 
+#ifdef HAVE_CONFIG_H
+# include <config.h>
+#endif
+
 /* OCC22218 NOTE: project dependency on gl2ps is specified by macro */
 #ifdef HAVE_GL2PS
   #include <gl2ps.h>
