Index: oce/BUILD.MINGWw64
===================================================================
--- /dev/null
+++ oce/BUILD.MINGWw64
@@ -0,0 +1,57 @@
+This will help you build OCE with MinGW w64/MSYS using the Oce-Win-Bundle sources:
+
+1) MinGW w64
+	Download MinGW w64 from http://mingw-w64.sourceforge.net/
+	Choose Mingw-builds Project, online Installer. (You can try others as well, but this one works...)
+	When Installing use POSIX as Threading Model and SEH as Exception Model (others might work too, but are untested)
+	Add the /bin directory of your MinGW install to your system's PATH-Variable (e.g. C:\mingw-builds\x64-4.8.1-posix-seh-rev5\mingw64\bin)
+	
+2) MSYS
+	Download MSYS from Mingw-builds Sourceforge Repository: 
+	http://sourceforge.net/projects/mingwbuilds/files/external-binary-packages/
+	Does not need installation, just extract into its own directory
+	
+3) CMake
+	Install CMake >2.8
+
+4) Open Cascade Community Edition
+	Obtain a local copy of the current OCE repository 
+		git clone git://github.com/tpaviot/oce.git
+			
+5) Oce-Win-Bundle
+	To obtain the FreeImage, FreeType and gl2ps libraries clone the oce-win-bundle repository into its own directory:
+		git clone https://github.com/QbProg/oce-win-bundle.git
+
+6) Run CMake
+    Point "Where is the source code:" to your local copy of OCE
+    Build the binaries in a new specific build folder within OCE
+    Hit Configure
+    Choose "MSYS Makefiles" (NOT "MinGW Makefiles"!) with default native compilers
+    When required: Point CMAKE_MAKE_PROGRAM to /bin/make.exe in your MSYS directory and configure again.
+    Check OCE_USE_BUNDLE_SOURCE
+    Check OCE_WITH_FREEIMAGE
+    Check OCE_WITH_GL2PS
+    Check OCE_VISUALISATION
+    Use OPENMP as OCE_MULTITHREAD_LIBRARY
+    Select your Installation directory with OCE_INSTALL_PREFIX
+    Configure
+    Select the directory of your oce-win-bundle clone with OCE_BUNDLE_ROOT_PATH
+    Select your required build type (RELEASE,DEBUG..)
+    Configure (and no entries should be red afterwards..)
+    and Generate.
+
+	
+7) Run MSYS and make
+	Start msys.bat in your MSYS-Directory (as an Administrator)
+	cd into your build directory
+	run 'make'
+	
+	Compilation is now running. It can take several hours, depending on your machine.
+	
+8) Install
+	run 'make install'
+	
+If you want to compile as DEBUG, you might have to compile twice, once as RELEASE and once as DEBUG to obtain the required release and debug libraries.
+If you do not need Visualisation, GL2PS etc. just uncheck them.
+
+Check for an updated Version of this File on the OCE Wiki on Github.
\ No newline at end of file
Index: oce/src/BRepMesh/BRepMesh_DiscretFactory.cxx
===================================================================
--- oce.orig/src/BRepMesh/BRepMesh_DiscretFactory.cxx
+++ oce/src/BRepMesh/BRepMesh_DiscretFactory.cxx
@@ -20,6 +20,10 @@
 #include <BRepMesh_IncrementalMesh.hxx>
 #include <BRepMesh_DiscretRoot.hxx>
 
+#ifdef HAVE_OCE_PATHS_H
+# include "oce-paths.h"
+#endif
+
 namespace
 {
   //! Embedded triangulation tool(s)
@@ -30,6 +34,14 @@ namespace
   static void MakeLibName (const TCollection_AsciiString& theDefaultName,
                                  TCollection_AsciiString& theLibName)
   {
+#ifdef OCE_LIBRARY_PREFIX
+    // Assemble library name according to the variables defined by CMAKE
+  	theLibName = "";
+  	theLibName += OCE_LIBRARY_PREFIX;
+  	theLibName += theDefaultName;
+  	theLibName += OCE_LIBRARY_DEBUG_POSTFIX;
+  	theLibName += OCE_LIBRARY_EXTENSION;
+#else
     theLibName = "";
   #ifndef WNT
     theLibName += "lib";
@@ -44,6 +56,7 @@ namespace
   #else
     theLibName += ".so";
   #endif
+#endif
   }
 };
 
Index: oce/src/Draw/Draw.cxx
===================================================================
--- oce.orig/src/Draw/Draw.cxx
+++ oce/src/Draw/Draw.cxx
@@ -503,6 +503,14 @@ void Draw::Load(Draw_Interpretor& theDI,
     }
 
     TCollection_AsciiString aPluginLibrary("");
+    
+#ifdef OCE_LIBRARY_PREFIX
+    // Assemble library name according to the variables defined by CMAKE
+    aPluginLibrary += OCE_LIBRARY_PREFIX;
+    aPluginLibrary += aPluginResource->Value(theKey.ToCString());
+    aPluginLibrary += OCE_LIBRARY_DEBUG_POSTFIX;
+    aPluginLibrary += OCE_LIBRARY_EXTENSION;
+#else
 #ifndef WNT
     aPluginLibrary += "lib";
 #endif
@@ -520,6 +528,8 @@ void Draw::Load(Draw_Interpretor& theDI,
 #else
     aPluginLibrary += ".so";
 #endif
+#endif
+
     OSD_SharedLibrary aSharedLibrary(aPluginLibrary.ToCString());
     if(!aSharedLibrary.DlOpen(OSD_RTLD_LAZY)) {
       TCollection_AsciiString error(aSharedLibrary.DlError());
Index: oce/src/NCollection/NCollection_UtfString.lxx
===================================================================
--- oce.orig/src/NCollection/NCollection_UtfString.lxx
+++ oce/src/NCollection/NCollection_UtfString.lxx
@@ -280,7 +280,7 @@ template<typename Type> inline
 void NCollection_UtfString<Type>::FromLocale (const char*            theString,
                                               const Standard_Integer theLength)
 {
-#if(defined(_WIN32) || defined(__WIN32__))
+  #ifdef _MSC_VER 
   // use WinAPI
   int aWideSize = MultiByteToWideChar (CP_ACP, MB_PRECOMPOSED, theString, -1, NULL, 0);
   if (aWideSize <= 0)
@@ -319,7 +319,7 @@ bool NCollection_UtfString<Type>::ToLoca
                                             const Standard_Integer theSizeBytes) const
 {
   NCollection_UtfString<wchar_t> aWideCopy (myString, myLength);
-#if(defined(_WIN32) || defined(__WIN32__))
+#ifdef _MSC_VER 
   int aMbBytes = WideCharToMultiByte (CP_ACP, 0, aWideCopy.ToCString(), -1, theBuffer, theSizeBytes, NULL, NULL);
 #else
   std::size_t aMbBytes = std::wcstombs (theBuffer, aWideCopy.ToCString(), theSizeBytes);
Index: oce/src/Plugin/Plugin.cxx
===================================================================
--- oce.orig/src/Plugin/Plugin.cxx
+++ oce/src/Plugin/Plugin.cxx
@@ -61,6 +61,14 @@ Handle(Standard_Transient) Plugin::Load
     }
     
     TCollection_AsciiString thePluginLibrary("");
+
+#ifdef OCE_LIBRARY_PREFIX
+    // Assemble library name according to the variables defined by CMAKE
+    thePluginLibrary += OCE_LIBRARY_PREFIX;
+    thePluginLibrary += PluginResource->Value(theResource.ToCString());
+    thePluginLibrary += OCE_LIBRARY_DEBUG_POSTFIX;
+    thePluginLibrary += OCE_LIBRARY_EXTENSION;
+#else
 #ifndef WNT
     thePluginLibrary += "lib";
 #endif
@@ -77,7 +85,9 @@ Handle(Standard_Transient) Plugin::Load
     thePluginLibrary += ".sl";
 #else
     thePluginLibrary += ".so";
-#endif  
+#endif
+#endif
+
     OSD_SharedLibrary theSharedLibrary(thePluginLibrary.ToCString());
     if(!theSharedLibrary.DlOpen(OSD_RTLD_LAZY)) {
       TCollection_AsciiString error(theSharedLibrary.DlError());
