Index: oce/src/Standard/Standard_CLocaleSentry.cxx
===================================================================
--- oce.orig/src/Standard/Standard_CLocaleSentry.cxx
+++ oce/src/Standard/Standard_CLocaleSentry.cxx
@@ -38,7 +38,7 @@ namespace
     CLocalePtr()
     #ifdef HAVE_XLOCALE_H
     : myLocale (newlocale (LC_ALL_MASK, "C", NULL))
-    #elif defined(_WIN32) && !defined(__BORLANDC__)
+    #elif defined(_WIN32) && !defined(__BORLANDC__) && !defined(__MINGW32__)
     : myLocale (_create_locale (LC_ALL, "C"))
     #else
     : myLocale (NULL)
@@ -49,7 +49,7 @@ namespace
     {
     #ifdef HAVE_XLOCALE_H
       freelocale (myLocale);
-    #elif defined(_WIN32) && !defined(__BORLANDC__)
+    #elif defined(_WIN32) && !defined(__BORLANDC__) && !defined(__MINGW32__)
       _free_locale (myLocale);
     #endif
     }
Index: oce/src/Standard/Standard_CLocaleSentry.hxx
===================================================================
--- oce.orig/src/Standard/Standard_CLocaleSentry.hxx
+++ oce/src/Standard/Standard_CLocaleSentry.hxx
@@ -56,7 +56,7 @@ public:
 
 #ifdef HAVE_XLOCALE_H
   typedef  locale_t clocale_t;
-#elif defined(_WIN32) && !defined(__BORLANDC__)
+#elif defined(_WIN32) && !defined(__BORLANDC__) && !defined(__MINGW32__)
   typedef _locale_t clocale_t;
 #else
   typedef void*     clocale_t;
@@ -69,7 +69,7 @@ public:
 private:
 
   void* myPrevLocale;       //!< previous locale, platform-dependent pointer!
-#if defined(_WIN32) && !defined(__BORLANDC__)
+#if defined(_WIN32) && !defined(__BORLANDC__) && !defined(__MINGW32__)
   int   myPrevTLocaleState; //!< previous thread-locale state, MSVCRT-specific
 #endif
 
Index: oce/src/Standard/Standard_CString.cxx
===================================================================
--- oce.orig/src/Standard/Standard_CString.cxx
+++ oce/src/Standard/Standard_CString.cxx
@@ -80,7 +80,7 @@ Standard_Integer HashCodes (const Standa
 #ifdef __APPLE__
   // There are a lot of *_l functions availalbe on Mac OS X - we use them
   #define SAVE_TL()
-#elif defined(_WIN32) && !defined(__BORLANDC__)
+#elif defined(_WIN32) && !defined(__BORLANDC__) && !defined(__MINGW32__)
   // MSVCRT has equivalents with slightly different syntax
   #define SAVE_TL()
   #define strtod_l(thePtr, theNextPtr, theLocale)                _strtod_l(thePtr, theNextPtr, theLocale)
