commit 81aec94ed4656a3fc4fc6394db597366a379b410
Author: QbProg <tholag@gmail.com>
Date:   Sun Jun 19 18:17:28 2011 +0200

    Fixed a buffer overlow in a local array

Index: oce/src/NIS/NIS_SurfaceDrawer.cxx
===================================================================
--- oce.orig/src/NIS/NIS_SurfaceDrawer.cxx
+++ oce/src/NIS/NIS_SurfaceDrawer.cxx
@@ -143,7 +143,7 @@
   glEnableClientState (GL_VERTEX_ARRAY);
   glEnable(GL_COLOR_MATERIAL);
 
-  Quantity_Parameter   aValueCol[3];
+  Quantity_Parameter   aValueCol[4];
   Quantity_TypeOfColor bidTC (Quantity_TOC_RGB);
   GLfloat aLineWidth (1.f);
   GLfloat anOffset = myPolygonOffset;
