Index: oce/src/BRepAlgo/BRepAlgo_Tool.cxx
===================================================================
--- oce.orig/src/BRepAlgo/BRepAlgo_Tool.cxx
+++ oce/src/BRepAlgo/BRepAlgo_Tool.cxx
@@ -87,7 +87,7 @@ TopoDS_Shape BRepAlgo_Tool::Deboucle3D(c
 	SubShape = Deboucle3D(CurS,Boundary);
 	if ( !SubShape.IsNull()) {
 	  B.Add(SS, SubShape);
-	  NbSub++;
+	  NbSub = 1;
 	}
       }
       if (NbSub == 0) SS = TopoDS_Shape();
Index: oce/src/Image/Image_GImage.gxx
===================================================================
--- oce.orig/src/Image/Image_GImage.gxx
+++ oce/src/Image/Image_GImage.gxx
@@ -901,11 +901,11 @@ void Image_GImage::DrawLine(const anyPix
 		  cout << "DrawLine( X and Y Axis Symetrie )\n" << flush ;
 #endif
 
-		  YAxisMirror++ ;
+		  YAxisMirror = 1 ;
 
 		  x1 = x0 - dx ;
 
-		  XAxisMirror++ ;
+		  XAxisMirror = 1 ;
 
 		  y1 = y0 - dy ;
 
@@ -924,7 +924,7 @@ void Image_GImage::DrawLine(const anyPix
 		  cout << "DrawLine( X Axis Symetrie )\n" << flush ;
 #endif
 
-		  XAxisMirror++ ;
+		  XAxisMirror = 1 ;
 
 		  y1 = y0 - dy ;
 
