From c8d429663f04cb127e4c955feb9c155ca8d1f6f1 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Sat, 2 Jun 2012 15:54:13 +0200
Subject: [PATCH] [warning-fix][comparison-between-different-enums]

---
 src/OpenGl/OpenGl_TextureBox.cxx |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: oce/src/OpenGl/OpenGl_TextureBox.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_TextureBox.cxx
+++ oce/src/OpenGl/OpenGl_TextureBox.cxx
@@ -400,7 +400,7 @@ static void MyGenTextureEXT (TextureID I
 static void MyBindTextureEXT (TextureID ID, int Context)
 {
   TextureDataID data = textab(ID).data;
-  if (texdata(data).status == TEX_NONE)
+  if (texdata(data).status == TEXDATA_NONE)
     return;
 
   GLenum aParamName = texdata(data).status == TEXDATA_1D ? 
