Index: oce/src/Standard/Standard_CString.cxx
===================================================================
--- oce.orig/src/Standard/Standard_CString.cxx
+++ oce/src/Standard/Standard_CString.cxx
@@ -89,10 +89,10 @@ Standard_Integer HashCode (const Standar
   static Standard_Integer Standard_Mask_String_Left[4] =
                   { 0 , 0x00ffffff , 0x0000ffff , 0x000000ff } ;
   static Standard_Integer Standard_Mask_String_Right[4] =
-                  { 0 , 0xff000000 , 0xffff0000 , 0xffffff00 } ;
+                  { 0 , (Standard_Integer) 0xff000000 , (Standard_Integer) 0xffff0000 , (Standard_Integer) 0xffffff00 } ;
 # else
   static Standard_Integer Standard_Mask_String_Left[4] =
-                  { 0 , 0xffffff00 , 0xffff0000 , 0xff000000 } ;
+                  { 0 , (Standard_Integer) 0xffffff00 , (Standard_Integer) 0xffff0000 , (Standard_Integer) 0xff000000 } ;
   static Standard_Integer Standard_Mask_String_Right[4] =
                   { 0 , 0x000000ff , 0x0000ffff , 0x00ffffff } ;
 # endif
