Index: oce/src/NCollection/NCollection_IncAllocator.cxx
===================================================================
--- oce.orig/src/NCollection/NCollection_IncAllocator.cxx
+++ oce/src/NCollection/NCollection_IncAllocator.cxx
@@ -90,7 +90,7 @@ Standard_EXPORT void IncAllocator_SetDeb
  * numbering of allocators.
  */
 //=======================================================================
-#ifdef _DEBUG
+#if defined(_DEBUG) && defined(DEB)
 static Standard_Size CurrentID = 0;
 static Standard_Size CATCH_ID = 0;
 
