From a2db37168aa07f4a10b4577d1ca227c5600e7ba2 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Tue, 19 Feb 2013 11:25:50 +0100
Subject: [PATCH] [clang-warning-fix][logical-op-parentheses]

---
 src/BRepOffsetAPI/BRepOffsetAPI_MiddlePath.cxx |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: oce/src/BOPAlgo/BOPAlgo_PaveFiller_6.cxx
===================================================================
--- oce.orig/src/BOPAlgo/BOPAlgo_PaveFiller_6.cxx
+++ oce/src/BOPAlgo/BOPAlgo_PaveFiller_6.cxx
@@ -1234,8 +1234,8 @@ static void ToleranceFF(const TopoDS_Fac
     const BOPDS_InterfEF& aEF = aEFs(i);
     nE = aEF.Index1();
     nFOpposite = aEF.Index2();
-    if(aMIF1.Contains(nE) && aMIF2.Contains(nFOpposite) ||
-       aMIF1.Contains(nFOpposite) && aMIF2.Contains(nE)) {
+    if((aMIF1.Contains(nE) && aMIF2.Contains(nFOpposite)) ||
+       (aMIF1.Contains(nFOpposite) && aMIF2.Contains(nE))) {
       //
       IntTools_CommonPrt aCP = aEF.CommonPart();
       if(aCP.Type() == TopAbs_VERTEX) {
@@ -1465,8 +1465,8 @@ static void ToleranceFF(const TopoDS_Fac
     const BOPDS_InterfVV& aVV = aVVs(i);
     nV1 = aVV.Index1();
     nV2 = aVV.Index2();
-    if(aMIF1.Contains(nV1) && aMIF2.Contains(nV2) ||
-       aMIF1.Contains(nV2) && aMIF2.Contains(nV1)) {
+    if((aMIF1.Contains(nV1) && aMIF2.Contains(nV2)) ||
+       (aMIF1.Contains(nV2) && aMIF2.Contains(nV1))) {
       if (aVV.HasIndexNew()) {
         nVNew = aVV.IndexNew();
         aMVStick.Add(nVNew);
@@ -1478,8 +1478,8 @@ static void ToleranceFF(const TopoDS_Fac
     const BOPDS_InterfVE& aVE = aVEs(i);
     nV = aVE.Index1();
     nE = aVE.Index2();
-    if(aMIF1.Contains(nV) && aMIF2.Contains(nE) ||
-       aMIF1.Contains(nE) && aMIF2.Contains(nV)) {
+    if((aMIF1.Contains(nV) && aMIF2.Contains(nE)) ||
+       (aMIF1.Contains(nE) && aMIF2.Contains(nV))) {
       aMVStick.Add(nV);
     }
   }
@@ -1488,8 +1488,8 @@ static void ToleranceFF(const TopoDS_Fac
     const BOPDS_InterfEE& aEE = aEEs(i);
     nE1 = aEE.Index1();
     nE2 = aEE.Index2();
-    if(aMIF1.Contains(nE1) && aMIF2.Contains(nE2) ||
-       aMIF1.Contains(nE2) && aMIF2.Contains(nE1)) {
+    if((aMIF1.Contains(nE1) && aMIF2.Contains(nE2)) ||
+       (aMIF1.Contains(nE2) && aMIF2.Contains(nE1))) {
       IntTools_CommonPrt aCP = aEE.CommonPart();
       if(aCP.Type() == TopAbs_VERTEX) {
         nVNew = aEE.IndexNew();
@@ -1502,8 +1502,8 @@ static void ToleranceFF(const TopoDS_Fac
     const BOPDS_InterfVF& aVF = aVFs(i);
     nV = aVF.Index1();
     nF = aVF.Index2();
-    if(aMIF1.Contains(nV) && aMIF2.Contains(nF) ||
-       aMIF1.Contains(nF) && aMIF2.Contains(nV)) {
+    if((aMIF1.Contains(nV) && aMIF2.Contains(nF)) ||
+       (aMIF1.Contains(nF) && aMIF2.Contains(nV))) {
       aMVStick.Add(nV);
     }
   }
@@ -1512,8 +1512,8 @@ static void ToleranceFF(const TopoDS_Fac
     const BOPDS_InterfEF& aEF = aEFs(i);
     nE = aEF.Index1();
     nF = aEF.Index2();
-    if(aMIF1.Contains(nE) && aMIF2.Contains(nF) ||
-       aMIF1.Contains(nF) && aMIF2.Contains(nE)) {
+    if((aMIF1.Contains(nE) && aMIF2.Contains(nF)) ||
+       (aMIF1.Contains(nF) && aMIF2.Contains(nE))) {
       IntTools_CommonPrt aCP = aEF.CommonPart();
       if(aCP.Type() == TopAbs_VERTEX) {
         nVNew = aEF.IndexNew();
Index: oce/src/BRepFill/BRepFill_TrimEdgeTool.cxx
===================================================================
--- oce.orig/src/BRepFill/BRepFill_TrimEdgeTool.cxx
+++ oce/src/BRepFill/BRepFill_TrimEdgeTool.cxx
@@ -357,10 +357,10 @@ void BRepFill_TrimEdgeTool::IntersectWit
   Standard_Real TolInit= 1.e-9;
   Standard_Integer nn = 7;
 
-  if(AC1.GetType() != GeomAbs_Circle &&
-     AC1.GetType() != GeomAbs_Line   ||
-     AC2.GetType() != GeomAbs_Circle &&
-     AC2.GetType() != GeomAbs_Line     ) {
+  if((AC1.GetType() != GeomAbs_Circle &&
+     AC1.GetType() != GeomAbs_Line)   ||
+     (AC2.GetType() != GeomAbs_Circle &&
+     AC2.GetType() != GeomAbs_Line)     ) {
 
     TolInit = 1.e-8;
     nn = 6;
Index: oce/src/BRepOffsetAPI/BRepOffsetAPI_MiddlePath.cxx
===================================================================
--- oce.orig/src/BRepOffsetAPI/BRepOffsetAPI_MiddlePath.cxx
+++ oce/src/BRepOffsetAPI/BRepOffsetAPI_MiddlePath.cxx
@@ -538,8 +538,8 @@ void BRepOffsetAPI_MiddlePath::Build()
       {
         anEdge = TopoDS::Edge(itl.Value());
         TopExp::Vertices(anEdge, V1, V2);
-        if ((V1.IsSame(PrevVertex) && V2.IsSame(CurVertex) ||
-             V1.IsSame(CurVertex) && V2.IsSame(PrevVertex)) &&
+        if (((V1.IsSame(PrevVertex) && V2.IsSame(CurVertex)) ||
+             (V1.IsSame(CurVertex) && V2.IsSame(PrevVertex))) &&
             !anEdge.IsSame(E1))
         {
           ProperEdge = anEdge;
@@ -854,8 +854,8 @@ void BRepOffsetAPI_MiddlePath::Build()
           }
           gp_Dir aDir1 = gp_Vec(aCirc.Location(), Pnt1);
           gp_Dir aDir2 = gp_Vec(aCirc.Location(), Pnt2);
-          if (!(aDir1.IsEqual(theDir1, AngTol) && aDir2.IsEqual(theDir2, AngTol) ||
-                aDir1.IsEqual(theDir2, AngTol) && aDir2.IsEqual(theDir1, AngTol)))
+          if (!((aDir1.IsEqual(theDir1, AngTol) && aDir2.IsEqual(theDir2, AngTol)) ||
+                (aDir1.IsEqual(theDir2, AngTol) && aDir2.IsEqual(theDir1, AngTol))))
           {
             SimilarArcs = Standard_False;
             break;
