Index: oce/src/BSplSLib/BSplSLib.cxx
===================================================================
--- oce.orig/src/BSplSLib/BSplSLib.cxx
+++ oce/src/BSplSLib/BSplSLib.cxx
@@ -39,7 +39,7 @@
 
 // for null derivatives
 static Standard_Real BSplSLib_zero[3] = {0.0, 0.0, 0.0};
-#ifdef WNT
+#if defined(WNT) && !defined(M_SQRT2)
 #define M_SQRT2 1.4142135623730950488016887 
 #endif
 
Index: oce/src/HLRBRep/HLRBRep_FaceData.lxx
===================================================================
--- oce.orig/src/HLRBRep/HLRBRep_FaceData.lxx
+++ oce/src/HLRBRep/HLRBRep_FaceData.lxx
@@ -19,7 +19,7 @@
 // and conditions governing the rights and limitations under the License.
 
 
-#define EMaskOrient     0x0000fu
+#define EMaskOrient_FD  0x0000fu
 #define FMaskSelected   0x00010u
 #define FMaskBack       0x00020u
 #define FMaskSide       0x00040u
@@ -303,7 +303,7 @@ inline void HLRBRep_FaceData::Size(const
 //=======================================================================
 
 inline TopAbs_Orientation HLRBRep_FaceData::Orientation() const
-{ return ((TopAbs_Orientation)(myFlags & EMaskOrient)); }
+{ return ((TopAbs_Orientation)(myFlags & EMaskOrient_FD)); }
 
 //=======================================================================
 //function : Orientation
@@ -312,8 +312,8 @@ inline TopAbs_Orientation HLRBRep_FaceDa
 
 inline void HLRBRep_FaceData::Orientation(const TopAbs_Orientation O)
 {
-  myFlags &= ~EMaskOrient;
-  myFlags |= (0 != (O & EMaskOrient));
+  myFlags &= ~EMaskOrient_FD;
+  myFlags |= (0 != (O & EMaskOrient_FD));
 }
 
 //=======================================================================
