Index: oce/src/CDF/CDF_FWOSDriver.cxx
===================================================================
--- oce.orig/src/CDF/CDF_FWOSDriver.cxx
+++ oce/src/CDF/CDF_FWOSDriver.cxx
@@ -167,7 +167,7 @@ TCollection_ExtendedString FWOSDriver_Dr
     
 #ifdef WNT
     TCollection_ExtendedString hd=UTL::xgetenv("HOMEDRIVE");
-    if(hd.Length() != NULL) {
+    if(hd.Length() != 0) {
       theDefaultFolder=hd;
       theDefaultFolder+=UTL::xgetenv("HOMEPATH");
     }
